src/backend/mi2/mi2.ts

```diff
@@ -25,27 +25,30 @@ function couldBeOutput(line: string) {
 const trace = false;
 
 export class MI2 extends EventEmitter implements IBackend {
-	constructor(public application: string, public preargs: string[], public extraargs: string[], procEnv: any, public extraCommands: string[] = []) {
+	constructor(public application: string, public preargs: string[], public extraargs: string[], procEnv: any, envFile: string, public extraCommands: string[] = []) {
 		super();
 
+		if (!(envFile || procEnv))
+			return;
+
+		// FIXME: if actuall debugged process is Win32, which we only know _after_ connect,
```

Comment:
Do we not really not know what type of process that will be debugged?  The environment support is only applicable to the debugger running on the same machine as the debug adapter (not supported over SSH), so it should be able to determine the system it's running on via `process.platform`.  Not sure I understand this issue.

Reply:
No, we don't know. I'm regularly using gdb to attach to gdbserver on a different machine myself. Works fine as long as the gdb has support for the target's architecture.

Reply:
But these environment variables are only being applied to the process where `gdb` is being run, not the process where `gdbserver` is run.  I don't think I understand what the expectation of the environment variables would be in a remote scenario like that.  Are they expected to be visible to the application being debugged?

Reply:
>  Are they expected to be visible to the application being debugged?

likely yes, as long as we use [`set environment`](https://sourceware.org/gdb/current/onlinedocs/gdb.html/Environment.html#set-environment) (or its mi equivalent) this will work both for GDB and (at least on GNU/Linux environments) gdbserver (since GDB 8.1); for running processes we`d need to call into `putenv()`, but I think it is fine to say that this is only supported for new inferiors.